scopus-import/src/main/java/no/sikt/nva/scopus/conversion/model/CorporationWithContributors.java

```diff
@@ -0,0 +1,148 @@
+        return isNull(cristinOrganizations)
+                   ? List.of()
+                   : cristinOrganizations;
+    }
+
+    public void setCristinOrganizations(List<CristinOrganization> cristinOrganizations) {
+        this.cristinOrganizations = cristinOrganizations;
+    }
+
+    public URI getCristinOrganizationId() {
+        return cristinOrganizationId;
+    }
+
+    public void setCristinOrganizationId(URI cristinId) {
+        this.cristinOrganizationId = cristinId;
+    }
+
+    public List<Corporation> toCorporations() {
+        return isNull(cristinOrganizations) || cristinOrganizations.isEmpty()
+                   ? generateCorporationFromAuthorGroupTp()
+                   : generateCorporationFromCristinOrganization();
+    }
+
+    public Builder copy() {
+        return new Builder().withAuthorGroupTp(scopusAuthors)
+                   .withCristinOrganizationId(cristinOrganizationId)
+                   .withCristinCorporations(cristinOrganizations);
+    }
+
+    private static boolean isNotNorway(Map<String, String> labels) {
+        return labels.values().stream().map(String::toLowerCase).noneMatch(NORWAY::equals);
+    }
+
+    private static String toOrganizationName(List<OrganizationTp> organizationTps) {
+        return organizationTps.stream()
+                   .map(organizationTp -> extractContentString(organizationTp.getContent()))
+                   .collect(Collectors.joining(AFFILIATION_DELIMITER));
+    }
+
+    private static String getIso6391LanguageCodeForSupportedNvaLanguage(String possiblyUnsupportedLanguageIso6391code) {
+        var language = LanguageMapper.getLanguageByIso6391Code(possiblyUnsupportedLanguageIso6391code);
+        return nonNull(language.getIso6391Code()) ? language.getIso6391Code() : ENGLISH.getIso6391Code();
+    }
+
+    private List<Corporation> generateCorporationFromCristinOrganization() {
+        return cristinOrganizations.stream()
+                   .map(cristinOrganization -> new Organization.Builder().withId(cristinOrganization.id()).build())
+                   .collect(Collectors.toList());
+    }
+
+    private List<Corporation> generateCorporationFromAuthorGroupTp() {
+        var name = getOrganizationNameFromAuthorGroup();
+        var labels = name.isPresent() && !name.get().isEmpty() ? name.map(
+            organizationName -> Map.of(getLanguageIso6391Code(organizationName), organizationName))
+                         : extractCountryNameAsAffiliation();
+        return isNotNorway(labels.orElse(Map.of()))
+                   ? List.of(new UnconfirmedOrganization(name.orElse(null)))
+                   : List.of();
+    }
+
+    private Optional<Map<String, String>> extractCountryNameAsAffiliation() {
+        return Optional.ofNullable(scopusAuthors.getAffiliation())
+                   .map(AffiliationTp::getCountry)
+                   .map(country -> Map.of(getLanguageIso6391Code(country), country));
```

Comment:
Country is a language code?

Reply:
No, but the country name might be "España", so we want the label to have "spa": "España".

Reply:
At least that was the intent of this code, it is from 2021 and just moved from somewhere else.